lib/elixir/lib/inspect/algebra.ex

```diff
@@ -88,8 +94,33 @@ defmodule Inspect.Opts do
           pretty: boolean,
           safe: boolean,
           syntax_colors: [{color_key, IO.ANSI.ansidata()}],
-          inspect_fun: (any, t -> Inspect.Algebra.t())
+          inspect_fun: (any, t -> Inspect.Algebra.t()),
+          custom_options: [{atom, any()}]
         }
+
+  def custom_options(opts) do
```

Comment:
@doc """
Filters out all non-default options.
"""

Reply:
It filters out all the _default_ options, leaving non-default only :)